jerry-core/ecma/base/ecma-gc.c

```diff
@@ -231,16 +231,30 @@ ecma_gc_mark_promise_object (ecma_extended_object_t *ext_object_p) /**< extended
 
   while (ecma_value_p != NULL)
   {
-    ecma_gc_set_object_visited (ecma_get_object_from_value (*ecma_value_p));
-    ecma_value_p = ecma_collection_iterator_next (ecma_value_p);
+    if (ecma_is_value_object (*ecma_value_p))
+    {
+      ecma_gc_set_object_visited (ecma_get_object_from_value (*ecma_value_p));
+      ecma_value_p = ecma_collection_iterator_next (ecma_value_p);
+    }
+    else
+    {
+      break;
+    }
   }
 
   ecma_value_p = ecma_collection_iterator_init (((ecma_promise_object_t *) ext_object_p)->reject_reactions);
 
   while (ecma_value_p != NULL)
   {
-    ecma_gc_set_object_visited (ecma_get_object_from_value (*ecma_value_p));
-    ecma_value_p = ecma_collection_iterator_next (ecma_value_p);
+    if (ecma_is_value_object (*ecma_value_p))
+    {
+      ecma_gc_set_object_visited (ecma_get_object_from_value (*ecma_value_p));
+      ecma_value_p = ecma_collection_iterator_next (ecma_value_p);
+    }
+    else
+    {
+      break;
```

Comment:
Aren't these callables? If yes, why don't we simply prevent adding non=objects to this collection?